### ISSUE TITLE
Don`t clear photo metadata

### ISSUE THREAD

**Issue by lpite** [localsend/localsend#237]
### Is your feature request related to a problem? Please describe.

Localsend clears all metadata from photos

### Describe the solution you'd like

-

**Comment by Tienisto** [comment]
Can you tell me what platforms you use?

Does this error still exist if "Save media to gallery" is turned off?

**Comment by lpite** [comment]
platform ios or windows.

if  "Save media to gallery" is turned off it saves metadata but if add to gallery date is not right

**Comment by wenLiangcan** [comment]
Same here. The images I sent to my iPhone from Windows by LocalSend had been reduced more than half the size and lost all the EXIF data.

**Comment by Tienisto** [comment]
I feel sorry for you...

I don't know a solution yet because "Save media to gallery" is handled by https://pub.dev/packages/image_gallery_saver and I don't know if there is a better library that actually respects EXIF data.

**Comment by wenLiangcan** [comment]
No problem and thanks for your work.


By having a look at the library you mentioned, may I suggest some advices? (Since I don't own a Mac, I'm afraid that I couldn't help on contributions of code)   

- [There](https://github.com/hui-z/image_gallery_saver/blob/master/lib/image_gallery_saver.dart#L15) is a default value 80 of the parameter `quality` of the method `saveImage`. So, the way you calling it currently would cause compression of the images.
- The `saveImage` method [converts](https://github.com/hui-z/image_gallery_saver/blob/master/ios/Classes/SwiftImageGallerySaverPlugin.swift#L26) image of any format to `JPEG`, I think this is the reason why metadata gets lost. 
- The way to keep the original media files is using the [`saveFile`](https://github.com/hui-z/image_gallery_saver/blob/master/lib/image_gallery_saver.dart#L29) method.

**Comment by Tienisto** [comment]
LocalSend already use saveFile:

https://github.com/localsend/localsend/blob/main/lib/util/file_saver.dart#L38

**Comment by wenLiangcan** [comment]
oops, sorry, I get you wrong

**Comment by TheGB0077** [comment]
This library says that it does no modifications to the source file, so maybe it's worth a look 🤔
https://pub.dev/packages/add_to_gallery

**Comment by Actuallymax** [comment]
same here, switch to landrop then everything is fine

**Comment by liuguanglin** [comment]
This bug still exists on the Android client V1.14.0.
I sent photos from Android to Windows, but the GPS information on these photos was lost.

**Comment by dvalter** [comment]
@liuguanglin this is an unrelated issue.

Android protects location privacy in a weird way, so that if you share via *media*, it would remove location EXIF data, **unless app had requested and was given a location permission**.

However if you share via *files*, it would share these files as they are without altering metadata.

**Comment by liuguanglin** [comment]
@dvalter 
But in the same Android environment, I can use the App **Snapdrop** to share photos with location EXIF data via media.

**Comment by chiak597** [comment]
Are you using the Android PhotoPicker? Maybe you are affected by this bug/feature:
https://issuetracker.google.com/issues/243294058

PhotoPicker ignores the permission and redacts the location metadata anyway. Google says "works as expected".

**Comment by liuguanglin** [comment]
I found that after re-granting it access to photos, it could share photos with location EXIF data via media.

**Comment by ShlomoCode** [comment]
@liuguanglin Can you elaborate more?

**Comment by liuguanglin** [comment]
@ShlomoCode Here is the general process: After installing LocalSend on Android, I launched it immediately and sent photos to the Windows client via the "Media" option. The app requested permission to access photos, but even when I granted it, the photos received on the Windows client would lose the location EXIF data.
Later, I revoked the app's permission to access photos and then re-granted it. After this re-authorization, when I sent photos again in the same manner, the location EXIF data was preserved.
